pkg/codegen/testing/test/testdata/plain-object-defaults/nodejs/types/mod1/input.ts

```diff
@@ -0,0 +1,24 @@
+// *** WARNING: this file was generated by test. ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+import * as pulumi from "@pulumi/pulumi";
+import * as inputs from "../../types/input";
+import * as outputs from "../../types/output";
+
+import * as utilities from "../utilities";
+
+/**
+ * A test for namespaces (mod 1)
+ */
+export interface TypArgs {
+    val?: pulumi.Input<string>;
+}
+/**
+ * typArgsProvideDefaults sets the appropriate defaults for TypArgs
+ */
+export function typArgsProvideDefaults(val: TypArgs): TypArgs {
```

Comment:
This is a concrete example of a non-type export in these inputs files. If we're still in doubt if re-export works as intended, we could check calling this engaging the re-export.

Reply:
Ah, the test `plain-object-defaults` does exactly this. I had a name error on this very type until very recently. 
I also removed the `export type` syntax in favor of `export * as ns` so I think its a nonissue.